src/main/java/io/stargate/sgv2/jsonapi/service/resolver/model/impl/CreateCollectionCommandResolver.java

```diff
@@ -279,18 +279,25 @@ private void validateAuthentication(
             .map(EmbeddingProvidersConfig.EmbeddingProviderConfig.TokenConfig::accepted)
             .toList();
 
-    // If the user hasn't provided authentication details, verify that the 'NONE' authentication
-    // type is enabled.
-    if (userConfig.authentication() == null) {
+    // If the user hasn't provided authentication details, verify that either the 'NONE' or 'HEADER'
```

Comment:
Verify according to the provider information from user' createCollection.
correct